dspy/clients/lm.py

```diff
@@ -53,38 +56,50 @@ def __init__(
                 max_tokens >= 5000 and temperature == 1.0
             ), "OpenAI's o1-* models require passing temperature=1.0 and max_tokens >= 5000 to `dspy.LM(...)`"
 
-    
-    
     async def acall(self, prompt=None, messages=None, **kwargs):
         """Async completion method"""
         cache = kwargs.pop("cache", self.cache)
         messages = messages or [{"role": "user", "content": prompt}]
         kwargs = {**self.kwargs, **kwargs}
 
         # Make the request and handle LRU & disk caching.
-        # NOTE(isaac): I removed caching
         if self.model_type == "chat":
-            completion = async_litellm_completion
+            completion = (
+                cached_async_litellm_completion if cache else async_litellm_completion
```

Comment:
I don't think this caching works yet as desired because we want async calls to get cached the same as non-async calls